### PR TITLE
Removes support for txt file for specifying initial admin password and prevents password from being printed in the logs

### PR DESCRIPTION
**Pull request by DarshitChanpura**
### Description
Storing an admin password in a text file that can be available to anyone that has access to config folder at any time could be potentially vulnerable. Although, originally added with the perception to support autonomous process to easily read from the config folder, we now only allow passing the admin password through environment variable as it is session-scoped. Further, we don't print the password to the logs to avoid any traces.

- Related to https://github.com/opensearch-project/security/issues/3847

### Testing
- automated tests

### Check List
- [x] New functionality includes testing
- [ ] New functionality has been documented
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by DarshitChanpura** [comment]
I'm updating the plugin install workflow rn, since it relied on text file being generated for passing admin password

File: src/main/java/org/opensearch/security/tools/democonfig/SecuritySettingsConfigurer.java [hunk 1]
```diff
@@ -34,6 +34,7 @@
 import org.opensearch.security.support.ConfigConstants;
 import org.opensearch.security.tools.Hasher;
 
+import joptsimple.internal.Strings;
```
Comment (cwperks) [inline src/main/java/org/opensearch/security/tools/democonfig/SecuritySettingsConfigurer.java:37]:
```suggestion
import org.opensearch.core.common.Strings;
```

Comment (derek-ho) [reply src/main/java/org/opensearch/security/tools/democonfig/SecuritySettingsConfigurer.java:37]: 👍

Comment (DarshitChanpura) [reply src/main/java/org/opensearch/security/tools/democonfig/SecuritySettingsConfigurer.java:37]: ah..good catch!

File: src/test/java/org/opensearch/security/tools/democonfig/SecuritySettingsConfigurerTests.java [hunk 2]
```diff
@@ -79,42 +76,25 @@ public void tearDown() throws NoSuchFieldException, IllegalAccessException {
...
-        securitySettingsConfigurer.updateAdminPassword();
-
-        assertEquals(customPassword, SecuritySettingsConfigurer.ADMIN_PASSWORD);
-        verifyStdOutContainsString("ADMIN PASSWORD SET TO: " + customPassword);
+        verifyStdOutContainsString("Admin password set successfully.");
     }
 
     @Test
     public void testUpdateAdminPassword_noPasswordSupplied() {
+        SecuritySettingsConfigurer.ADMIN_PASSWORD = ""; // to ensure 0 flaky-ness
```
Comment (derek-ho) [inline src/test/java/org/opensearch/security/tools/democonfig/SecuritySettingsConfigurerTests.java:97]: Why is this necessary?

Comment (DarshitChanpura) [reply src/test/java/org/opensearch/security/tools/democonfig/SecuritySettingsConfigurerTests.java:97]: flkay-ness caused probly due to static variable.

Comment (DarshitChanpura) [reply src/test/java/org/opensearch/security/tools/democonfig/SecuritySettingsConfigurerTests.java:97]: in my local setup, I saw this error once in about 5 runs. After adding this fix I was able to verify that it is no longer flaky. (ran about 20 times locally)

Comment (DarshitChanpura) [reply src/test/java/org/opensearch/security/tools/democonfig/SecuritySettingsConfigurerTests.java:97]: In actual run, this static variable will be assigned only once, but for testing it maybe assigned multiple times which can cause flaky-ness.

**Review by derek-ho — APPROVED** [review]
LGTM

File: src/test/java/org/opensearch/security/tools/democonfig/SecuritySettingsConfigurerTests.java (hunk 2 above)
Comment (DarshitChanpura) [reply src/test/java/org/opensearch/security/tools/democonfig/SecuritySettingsConfigurerTests.java:97]: here is an example run: https://github.com/opensearch-project/security/actions/runs/7213760630/job/19656615873?pr=3844

File: src/main/java/org/opensearch/security/tools/democonfig/SecuritySettingsConfigurer.java [hunk 3]
```diff
@@ -107,21 +130,10 @@ void updateAdminPassword() {
                     .build()
             );
 
-            // Read custom password
+            // Read custom password from environment variable
+            String initialAdminPassword = System.getenv().get(ConfigConstants.OPENSEARCH_INITIAL_ADMIN_PASSWORD);
             if (initialAdminPassword != null && !initialAdminPassword.isEmpty()) {
```
Comment (cwperks) [inline src/main/java/org/opensearch/security/tools/democonfig/SecuritySettingsConfigurer.java:135]: nit: Use `Strings.inNullOrEmpty` here.

**Review by cwperks — APPROVED** [review]

File: src/main/java/org/opensearch/security/tools/democonfig/SecuritySettingsConfigurer.java (hunk 3 above)
Comment (DarshitChanpura) [reply src/main/java/org/opensearch/security/tools/democonfig/SecuritySettingsConfigurer.java]: done.

**Review by cwperks — APPROVED** [review]

**Review by peternied — APPROVED** [review]
